apps/language_server/test/dialyzer_test.exs

```diff
@@ -113,22 +113,35 @@ defmodule ElixirLS.LanguageServer.DialyzerTest do
 
         c_uri = SourceFile.path_to_uri("lib/c.ex")
 
+        assert_receive notification("window/logMessage", %{
+                         "message" => "[ElixirLS Dialyzer] Found " <> _
+                       })
+
+        assert_receive notification("window/logMessage", %{
+                         "message" => "[ElixirLS Dialyzer] Done writing manifest" <> _
+                       }),
+                       3_000
+
+        Server.receive_packet(server, did_open(c_uri, "elixir", 1, c_text))
+
         # The dialyzer process checks a second back since mtime only has second
-        # granularity, so we need to trigger one save that will set the
-        # timestamp, and then wait a second and save again to get the actual
-        # changed file.
-        for _ <- 1..2 do
-          Server.receive_packet(server, did_open(c_uri, "elixir", 1, c_text))
-          File.write!("lib/c.ex", c_text)
-          Server.receive_packet(server, did_save(c_uri))
-          assert_receive publish_diagnostics_notif(^file_c, []), 20_000
-
-          Process.sleep(1_500)
-        end
+        # granularity, so we need to wait a second.
+
+        File.write!("lib/c.ex", c_text)
+        Process.sleep(1_500)
+        Server.receive_packet(server, did_save(c_uri))
 
         assert_receive notification("window/logMessage", %{
-                         "message" => "[ElixirLS Dialyzer] Found 1 changed files" <> _
```

Comment:
The assumptions in the assert was wrong. Due to protocol consolidation purging all protocol related beam files are also changed